File: dart-jaguar/lib/model/response_time_filter_postcode_sector_properties.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';


import 'package:openapi/model/response_travel_time_statistics.dart';

part 'response_time_filter_postcode_sector_properties.jser.dart';

class ResponseTimeFilterPostcodeSectorProperties {
  
  @Alias('travel_time_reachable', isNullable: false,  )
  final ResponseTravelTimeStatistics travelTimeReachable;
  
  @Alias('travel_time_all', isNullable: false,  )
  final ResponseTravelTimeStatistics travelTimeAll;
  
  @Alias('coverage', isNullable: false,  )
  final double coverage;
  

  ResponseTimeFilterPostcodeSectorProperties(
      

{
     this.travelTimeReachable = null,  
     this.travelTimeAll = null,  
     this.coverage = null 
    
    }
  );

  @override
  String toString() {
    return 'ResponseTimeFilterPostcodeSectorProperties[travelTimeReachable=$travelTimeReachable, travelTimeAll=$travelTimeAll, coverage=$coverage, ]';
  }
}

@GenSerializer(nullableFields: true)
class ResponseTimeFilterPostcodeSectorPropertiesSerializer extends Serializer<ResponseTimeFilterPostcodeSectorProperties> with _$ResponseTimeFilterPostcodeSectorPropertiesSerializer {

}

File: fsharp-functions/OpenAPI/src/model/ResponseTimeFilterPostcodeSectorProperties.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.ResponseTravelTimeStatistics

module ResponseTimeFilterPostcodeSectorProperties = 

  //#region ResponseTimeFilterPostcodeSectorProperties

  [<CLIMutable>]
  type ResponseTimeFilterPostcodeSectorProperties = {
    [<JsonProperty(PropertyName = "travel_time_reachable")>]
    TravelTimeReachable : ResponseTravelTimeStatistics;
    [<JsonProperty(PropertyName = "travel_time_all")>]
    TravelTimeAll : ResponseTravelTimeStatistics;
    [<JsonProperty(PropertyName = "coverage")>]
    Coverage : double;
  }
  
  //#endregion
  